x-pack/plugins/observability/server/services/slo/transform_generators/transform_generator.ts

```diff
@@ -30,12 +32,20 @@ export abstract class TransformGenerator {
           source: `emit(${slo.revision})`,
         },
       },
-      'slo.instanceId': {
+      'slo.groupBy': {
         type: 'keyword',
         script: {
-          source: `emit('${ALL_VALUE}')`,
+          source: `emit('${!!slo.groupBy ? slo.groupBy : ALL_VALUE}')`,
         },
       },
+      ...(mustIncludeAllInstanceId && {
+        'slo.instanceId': {
+          type: 'keyword',
+          script: {
+            source: `emit('${ALL_VALUE}')`,
+          },
+        },
+      }),
```

Comment:
💬 We always include the `slo.groupBy` field to the runtime mappings, but we only include the `slo.instanceId` when the groupBy key is `*` or empty `""`, otherwise we rely on the terms aggregation to get the real instanceId values.